middleware/primary_resource_logic/data_requests.py

```diff
@@ -287,20 +290,34 @@ def get_data_request_related_sources(db_client: DatabaseClient, dto: GetByIDBase
         role=RelationRoleEnum.STANDARD,
         column_permission=ColumnPermissionEnum.READ,
     )
-    results = db_client.get_linked_rows(
-        link_table=Relations.LINK_DATA_SOURCES_DATA_REQUESTS,
-        left_id=int(dto.resource_id),
-        left_link_column="request_id",
-        right_link_column="source_id",
-        linked_relation=Relations.DATA_SOURCES_EXPANDED,
-        linked_relation_linking_column="airtable_uid",
-        columns_to_retrieve=permitted_columns,
+    results = db_client.get_data_requests(
+        columns=["id"],
+        where_mappings=[WhereMapping(column="id", value=int(dto.resource_id))],
+        subquery_parameters=[
+            SubqueryParameterManager.get_subquery_params(
+                relation=Relations.DATA_SOURCES_EXPANDED,
+                linking_column="data_sources",
+                columns=permitted_columns,
+            )
+        ],
+        build_metadata=True,
     )
+
+    metadata = results["metadata"]
```

Comment:
NIT: In a case like this, probably a good idea to extract the logic to a separate function, just to keep things concise and at the same level of abstraction within the function.